server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/ReindexingRoutesTest.java

```diff
@@ -0,0 +1,708 @@
+
+import io.restassured.RestAssured;
+
+class ReindexingRoutesTest {
+    private static final String USERNAME = "benwa@apache.org";
+    private static final MailboxPath INBOX = MailboxPath.forUser(USERNAME, "INBOX");
+
+    private WebAdminServer webAdminServer;
+    private ListeningMessageSearchIndex searchIndex;
+    private InMemoryMailboxManager mailboxManager;
+
+    @BeforeEach
+    void beforeEach() throws Exception {
+        mailboxManager = new InMemoryIntegrationResources().createMailboxManager(new SimpleGroupMembershipResolver());
+        MemoryTaskManager taskManager = new MemoryTaskManager();
+        InMemoryId.Factory mailboxIdFactory = new InMemoryId.Factory();
+        searchIndex = mock(ListeningMessageSearchIndex.class);
+        ReIndexer reIndexer = new ReIndexerImpl(
+            new ReIndexerPerformer(
+                mailboxManager,
+                searchIndex,
+                mailboxManager.getMapperFactory()));
+        JsonTransformer jsonTransformer = new JsonTransformer();
+
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new DefaultMetricFactory(),
+            new TasksRoutes(taskManager, jsonTransformer),
+            new ReindexingRoutes(
+                taskManager,
+                mailboxManager,
+                mailboxIdFactory,
+                reIndexer,
+                jsonTransformer));
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer).build();
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    }
+
+    @Nested
+    class FullReprocessing {
+        @Nested
+        class Validation {
+            @Test
+            void fullReprocessingShouldFailWithNoTask() {
+                when()
+                    .post("/index")
+                .then()
+                    .statusCode(HttpStatus.BAD_REQUEST_400)
+                    .body("statusCode", is(400))
+                    .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                    .body("message", is("task query parameter is mandatory. The only supported value is `reIndex`"));
+            }
+
+            @Test
+            void fullReprocessingShouldFailWithBadTask() {
+                when()
+                    .post("/index?task=bad")
+                .then()
+                    .statusCode(HttpStatus.BAD_REQUEST_400)
+                    .body("statusCode", is(400))
+                    .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                    .body("message", is("task query parameter is mandatory. The only supported value is `reIndex`"));
+            }
+        }
+
+        @Nested
+        class TaskDetails {
+            @Test
+            void fullReprocessingShouldNotFailWhenNoMail() {
+                String taskId = with()
+                    .post("/index?task=reIndex")
```

Comment:
why this API? it sounds strange to me, but I'm not very restful.

Reply:
It's what we did in the MailRepository reprocessing.